near-sdk/compilation_tests/generic_function.rs

```diff
@@ -0,0 +1,19 @@
+//! Functions can't use generics.
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_sdk::near_bindgen;
+
+#[near_bindgen]
+#[derive(Default, BorshDeserialize, BorshSerialize)]
+struct Ident {
+    value: u32,
+}
+
+#[near_bindgen]
+impl Ident {
+    pub fn is_ident<T>(&self, val: T) -> T {
+        val
+    }
```

Comment:
Once the logic is extended to disallow const generics, lets add a case here.

Reply:
Created a separate test case so that we can check the error message (it only reports one at a time, probably because the macro errors work differently from regular compilation errors).

Reply:
.. should resolve this issue

Reply:
Unfortunately, resolving this issue would require doing the same refactoring in `ItemImplInfo`, `ItemTraitInfo` and potentially some other places. I think this is better done in a separate PR.

Reply:
Ahh, right! That's unfortunate.. combining errors doesn't apply then, since `AttrSigInfo` is called per-method signature. But hey, at least we get good spans 😅.

But refactoring the macro system is in order tbh.